cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/core/service/worker/impl/CoordinateCleanerImpl.java

```diff
@@ -0,0 +1,122 @@
+import edu.asu.diging.cord19.explorer.core.model.impl.WikipediaArticleImpl;
+
+@Service
+@PropertySource({ "classpath:config.properties", "${appConfigFile:classpath:}/app.properties" })
+public class CoordinateCleanerImpl implements CoordinateCleaner {
+
+    @Autowired
+    private TaskRepository taskRepo;
+
+    @Autowired
+    private PublicationRepository pubRepo;
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+
+    @Autowired
+    private MapTotalsRepository totalsRepo;
+    
+
+    public void calculateCountryStats() {
+
+        MapTotalsImpl totals = new MapTotalsImpl();
+        try (CloseableIterator<CountriesImpl> countries = mongoTemplate.stream(new Query().noCursorTimeout(),
+                CountriesImpl.class)) {
+            while (countries.hasNext()) {
+                CountriesImpl country = countries.next();
+                if (country.getProperties().getSelectedWikipediaCount() > 0) {
+                    if (country.getProperties().getSelectedWikipediaCount() > totals.getHighCount()) {
+                        totals.setHighCount(country.getProperties().getSelectedWikipediaCount());
+                    }
+                    if (country.getProperties().getSelectedWikipediaCount() < totals.getLowCount()) {
+                        totals.setLowCount(country.getProperties().getSelectedWikipediaCount());
+                    }
+                }
+            }
+        }
+
+        totalsRepo.save(totals);
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see edu.asu.diging.cord19.explorer.core.service.worker.impl.DocumentCleaner#
+     *
+     */
+    @Override
+    @Async
+    public void startCleaningCoordinates(String taskId) {
+
+        Optional<TaskImpl> optional = taskRepo.findById(taskId);
+        if (!optional.isPresent()) {
+            return;
+        }
+
+        Task task = optional.get();
+        task.setStatus(TaskStatus.PROCESSING);
+        taskRepo.save((TaskImpl) task);
+        
+        CoordinateParser coordinateParser = new CoordinateParser();
+
+        try (CloseableIterator<PublicationImpl> docs = mongoTemplate.stream(new Query(), PublicationImpl.class)) {
+            while (docs.hasNext()) {
+                PublicationImpl pub = docs.next();
+                List<PersonImpl> authors = pub.getMetadata().getAuthors();
+                for (PersonImpl author : authors) {
+                    if (author.getAffiliation().getSelectedWikiarticle() != null) {
+                        WikipediaArticleImpl article = author.getAffiliation().getSelectedWikiarticle();                        
+                        article.setCleanedCoords(coordinateParser.parse(article.getCoordinates()));
+                    }
+                }
+                pubRepo.save(pub);
+            }
+        }
+
+        calculateCountryStats();
+
+        task.setStatus(TaskStatus.DONE);
+        taskRepo.save((TaskImpl) task);
+    }
+
+    public Publication cleanCoordinatesImport(Publication pub) {
```

Comment:
why is this called `cleanCoordinatesImport` and not just `cleanCoordinates`? Is the import part important?